eng/pipelines/templates/steps/upload-failure-logs.yml

```diff
@@ -0,0 +1,29 @@
+parameters:
+  Condition: failed()
+  SourceFolder: $(Build.SourcesDirectory)
+  ContentExpression: /**/build/**/*.log
+  ArtifactName: BuildFailureLogs_$(Agent.JobName)_$(System.JobAttempt)
+
+steps:
+  - task: CopyFiles@2
+    condition: ${{ parameters.Condition }}
+    displayName: Stage failure logs for upload
+    inputs:
+      SourceFolder: $(Build.SourcesDirectory)
+      Contents: '${{ parameters.ContentExpression }}'
+      TargetFolder: $(Build.ArtifactStagingDirectory)/failurelogs
+
+  - task: PublishPipelineArtifact@1
```

Comment:
https://github.com/Azure/azure-sdk-for-cpp/blob/main/eng/common/pipelines/templates/steps/publish-artifact.yml

You want to use that template otherwise retries on failure will be broken.

Reply:
I think any benefits from `publish-artifacts.yml` can also be accomplished by hardcoding `${{ parameters.ArtifactName }}_$(System.JobAttempt)` into the `inputs.artifactName` property and not giving the calling pipeline the opportunity to make that mistake. 

Another side effect of using the `publish-artifact.yml` template is that there's always a redundant task added in that will never be run because the condition would work out to `and(succeeded(), failed())`

Reply:
After discussion we're taking a different approach and outputting a narrow set of logs to the console. Any sensitive information will be known to DevOps and will be redacted from those logs.